lib/tasks/convert_embeddables.rake

```diff
@@ -303,28 +303,31 @@ def delete_orphaned_old_embeddables()
 end
 
 def replace_old_embeddables(activity_id=nil)
-  serialization_helper = LaraSerializationHelper.new
-  where_conditions = activity_id ? {id: activity_id} : {runtime: "LARA"}
-  page_items = PageItem.includes(interactive_page: {lightweight_activity: []}).where(embeddable_type: @convertable_embeddable_classes).where(lightweight_activities: where_conditions)
-  page_items_processed = 0
-  page_items_to_process = page_items.count
-  page_items.find_each(batch_size: 100) do |page_item|
-    new_embeddable = ManagedInteractive.where(legacy_ref_id: page_item.embeddable_id, legacy_ref_type: page_item.embeddable_type).first
-    if new_embeddable
-      page_item.embeddable = new_embeddable
-      page_item.save!
-      wrapping_plugins = Embeddable::EmbeddablePlugin.where(embeddable_id: new_embeddable.legacy_ref_id, embeddable_type: new_embeddable.legacy_ref_type)
-      wrapping_plugins.each do |wrapping_plugin|
-        wrapping_plugin.embeddable = new_embeddable
-        wrapping_plugin.save!
-      end
-      page_items_processed += 1
-      if page_items_processed % 100 == 0
-        puts "Processed #{page_items_processed} page items."
-      end
+  # TODO: Preload wrapping plugins in main query before looping through embeddables.
```

Comment:
Something you can try is to add:
`has_many :embeddable_plugins, as: :embeddable`
To each of the legacy embeddables.
Then you might be able to do:
`new_embeddables = ManagedInteractive.includes(:page_item).preload(converted_plugins: [], converted_interactive: [:embeddable_plugins, {page_items: {interactive_page: [:lightweight_activity]}}]}).where("legacy_ref_id IS NOT NULL AND legacy_ref_id != ?", "").where("page_items.id IS NULL")`